src/ngx_http_modsecurity_module.c

```diff
@@ -727,6 +744,14 @@ ngx_http_modsecurity_merge_conf(ngx_conf_t *cf, void *parent, void *child)
         return strdup(error);
     }
 
+    // This is necessary for logs to be reopened when the config is reloaded
+    if (msc_rules_reopen_audit_log(c->rules_set, &error) < 0) {
```

Comment:
I reintroduced the `reopen()` method for Writers, so we should no longer see these issues. Since it looks like ModSecurity either supports or wants to support custom loggers, it makes sense to have a logger-specific `reopen()` method anyway.

https://github.com/brandonpayton/ModSecurity/pull/2/commits/708e3a3682681cce0b4f01795202015716ec5fcb

Reply:
I tested with `SecAuditEngine` off while `SecAuditLog` and `SecAuditLog2` were populated, and there were no issues loading the configuration.

Reply:
that is, after the change to make log writers responsible for reopen, there are no issues loading the above configuration.